VideoLocker/src/main/java/org/edx/mobile/task/GetEnrolledCourseTask.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * GetEnrolledCourseTask
+ *
+ * This task is used to get the course information for a single course from the api which does not
+ * have an endpoint to retrieve data for a single course. It uses the endpoint that returns a list
+ * of enrolled courses and then parses out the course that matches the given course id.
+ */
+
+package org.edx.mobile.task;
+
+import android.content.Context;
+
+import org.edx.mobile.model.api.EnrolledCoursesResponse;
+import org.edx.mobile.services.ServiceManager;
+
+import java.util.ArrayList;
+
+public abstract class GetEnrolledCourseTask extends Task<EnrolledCoursesResponse> {
+
+    String courseId;
+    ArrayList<EnrolledCoursesResponse> enrolledCoursesResponse;
+
+    public GetEnrolledCourseTask(Context context, String courseId) {
+        super(context);
+        this.courseId = courseId;
+    }
+
+    @Override
+    public EnrolledCoursesResponse call( ) throws Exception{
+        try {
+            if(courseId!=null){
+                ServiceManager api = environment.getServiceManager();
+                try {
+                    enrolledCoursesResponse = new ArrayList<EnrolledCoursesResponse>(api.getEnrolledCourses(false));
+                    for (EnrolledCoursesResponse course : enrolledCoursesResponse) {
+                        if (course.getCourse().getId().equals(courseId)) {
+                            return course;
+                        }
+                    }
+                } catch(Exception ex) {
+                    logger.error(ex, true);
+                }
+            }
+            return null;
+        } catch(Exception ex) {
```

Comment:
I would rather we let the Exception bubble out of this method; If you want to handle the exception, do it at the call site where this task is created.